Reskinned Domain Picker: Fix content shifts on medium-sized viewports

StefanNieuwenhuis opened the pull request:
### Changes proposed in this Pull Request

* On medium-sized viewports, the content of the Domain step is centered.
* The content always take full width of the content area (max-width being set to 1280px)

### Testing instructions
1. Check out this branch.
2. `yarn && yarn start`
3. Navigate to `/start/domains?flags=signup/reskin`
4. Open dev tools and go through all breakpoints (set the viewport to responsive and resize the screen).
5. Ensure that:
- [ ] The page content is centered on medium-sized viewports.
- [ ] The side content blocks are not shrunken (i.e. condensed) on medium-sized viewports.

Fixes #53936

StefanNieuwenhuis committed:
Initial commit

StefanNieuwenhuis committed:
update flex props on medium viewports

- justify-content: center;

StefanNieuwenhuis committed:
update title & subtitle alignment

- text-align: center on large breakpoints.

StefanNieuwenhuis commented on client/signup/steps/domains/index.jsx:
```diff
@@ -812,7 +812,6 @@ class DomainsStep extends React.Component {
 				goToNextStep={ this.handleSkip }
 				skipHeadingText={ translate( 'Not sure yet?' ) }
 				skipLabelText={ translate( 'Choose a domain later' ) }
-				align={ isReskinned ? 'left' : 'center' }
```
This is redundant (hence removed) since the alignment for both title & subtitle is defined in the css.

StefanNieuwenhuis replied:
Reverted.

StefanNieuwenhuis commented on client/signup/steps/domains/style.scss:
```diff
@@ -54,6 +54,10 @@ body.is-section-signup.is-white-signup {
 	.domains__step-content {
 		display: flex;
 
+		@include break-small {
+			justify-content: center;
+		}
```
Since the login & plans (sub)titles are centered on medium-sized viewports onwards (see Figma gnYBUOBthTMckJRJr3pZmk-fi-2624%3A237, gnYBUOBthTMckJRJr3pZmk-fi-2572%3A0 & gnYBUOBthTMckJRJr3pZmk-fi-2572%3A124), I applied it to the domain picker as well.

StefanNieuwenhuis commented on client/signup/style.scss:
```diff
@@ -811,10 +811,18 @@ body.is-section-signup.is-white-signup .layout:not( .dops ):not( .is-wccom-oauth
 		.formatted-header__subtitle {
 			text-align: left;
+
+			@include break-large {
+				text-align: center;
+			}
```
Since the login & plans (sub)titles are centered on medium-sized viewports onwards (see Figma gnYBUOBthTMckJRJr3pZmk-fi-2624%3A237, gnYBUOBthTMckJRJr3pZmk-fi-2572%3A0 & gnYBUOBthTMckJRJr3pZmk-fi-2572%3A124), I applied it to the domain picker as well.

razvanpapadopol replied:
That wasn't a bug. Domains step layout is slightly different from the design. See figma: gnYBUOBthTMckJRJr3pZmk-fi-2767%3A18

StefanNieuwenhuis commented on client/signup/style.scss:
```diff
@@ -811,10 +811,18 @@ body.is-section-signup.is-white-signup .layout:not( .dops ):not( .is-wccom-oauth
 		.formatted-header__title {
 			text-align: left;
+
+			@include break-large {
+				text-align: center;
+			}
```
See https://github.com/Automattic/wp-calypso/pull/53938/files#r656922043

StefanNieuwenhuis replied:
Reverted.

StefanNieuwenhuis committed:
update width of side content blocks

- the width changes to 20vw on break-xlarge.

StefanNieuwenhuis committed:
Revert "update flex props on medium viewports"

This reverts commit b800681da5b838d84f8f10a9d2f0cb297164f211.

StefanNieuwenhuis committed:
Revert "update width of side content blocks"

This reverts commit 2fc91172f7c2394f9319c5e74a47252e8984528e.

StefanNieuwenhuis committed:
Revert "update title & subtitle alignment"

This reverts commit 4dddbe77ed800bed09c34a614439950b6879f712.

StefanNieuwenhuis committed:
update flex props search bar + results container

- set flex: 1; to make the container grow and shrink when necessary.

StefanNieuwenhuis committed:
update margins for side content blocks

We changed the margins on the viewports in order to keep the content well-aligned and in screen.

StefanNieuwenhuis committed:
update max width for results + sidebar container on large screens

StefanNieuwenhuis committed:
update max width for results + sidebar container on large screens

StefanNieuwenhuis committed:
update search bar margins

On medium-sized viewports, the search bar is pushed down. This is solved by changing the margin-top in the viewport.

razvanpapadopol reviewed (changes requested):
While the layout looks good on medium-sized viewports, there is still one improvement that can be done: set an absolute value of `350px` [to the width of the side column on large screens here](https://github.com/Automattic/wp-calypso/blob/trunk/client/signup/steps/domains/style.scss#L87) to keep the proportions as in Figma.

✅ Good proportions on 1024px wide viewport

<img width="600" alt="Screenshot 2021-06-24 at 10 29 36" src="https://user-images.githubusercontent.com/14192054/123236073-23505a80-d4e5-11eb-9879-83b0d6291d99.png">

❌ Side column is too small on 1440px viewport

<img width="600" alt="Screenshot 2021-06-24 at 10 30 10" src="https://user-images.githubusercontent.com/14192054/123236332-614d7e80-d4e5-11eb-9e47-ad4e76f83834.png">

❌ Side column is too wide on 4K monitors

<img width="600" alt="Screenshot 2021-06-24 at 10 29 08" src="https://user-images.githubusercontent.com/14192054/123236369-690d2300-d4e5-11eb-876a-0cbc46afb240.png">

StefanNieuwenhuis committed:
update side-content container blocks width & margins

StefanNieuwenhuis commented:
> While the layout looks good on medium-sized viewports, there is still one improvement that can be done: set an absolute value of `350px` [to the width of the side column on large screens here](https://github.com/Automattic/wp-calypso/blob/trunk/client/signup/steps/domains/style.scss#L87) to keep the proportions as in Figma.
> 
> ✅ Good proportions on 1024px wide viewport
> 
> <img alt="Screenshot 2021-06-24 at 10 29 36" width="600" src="https://user-images.githubusercontent.com/14192054/123236073-23505a80-d4e5-11eb-9879-83b0d6291d99.png">
> 
> ❌ Side column is too small on 1440px viewport
> 
> <img alt="Screenshot 2021-06-24 at 10 30 10" width="600" src="https://user-images.githubusercontent.com/14192054/123236332-614d7e80-d4e5-11eb-9e47-ad4e76f83834.png">
> 
> ❌ Side column is too wide on 4K monitors
> 
> <img alt="Screenshot 2021-06-24 at 10 29 08" width="600" src="https://user-images.githubusercontent.com/14192054/123236369-690d2300-d4e5-11eb-876a-0cbc46afb240.png">

Thanks @razvanpapadopol!

I've changed the width and I've updated the margins as well to prevent strange jumps (see [video](https://www.loom.com/share/65c516c9ee004a849172324adf4fb708)). Can you have another look?

razvanpapadopol reviewed (approved):
Thanks, @StefanNieuwenhuis ! 🚀
